obspy/io/gcf/core.py

```diff
@@ -41,60 +175,673 @@ def _is_gcf(filename):
+        # File either does not point at a file object or file is not of
+        # proper size
+        is_gcf = False
+    else:
+        # Load shared library
+        gcf_io = _load_cdll("gcf")
+
+        # declare function argument and return types
+        gcf_io.read_gcf.argtypes = [ctypes.c_char_p,
+                                    ctypes.POINTER(_GcfFile),
+                                    ctypes.c_int]
+        gcf_io.read_gcf.restype = ctypes.c_int
+        gcf_io.free_GcfFile.argtypes = [ctypes.POINTER(_GcfFile)]
+        gcf_io.free_GcfFile.restype = None
 
+        # Decode first block
+        obj = _GcfFile()
+        b_filename = filename.encode('utf-8')
+        ret = gcf_io.read_gcf(b_filename, obj, 3)
+        if ret or (obj.n_errHead and obj.seg[0].err not in (10, 11, 21)) \
+                or obj.n_errData:
+            is_gcf = False
+            if filename[-4:] == ".gcf":
+                print("file: %s" % (filename))
+                print("ret: %d" % (ret))
+                print("n-blocks: %d" % os.path.getsize(filename)/1024.)
+                print("obj.n_blk = %d" % (obj.n_blk))
+                print("obj.n_errHead = %d" % (obj.n_errHead))
+                print("obj.n_errData = %d" % (obj.n_errData))
 
-def _read_gcf(filename, headonly=False, **kwargs):  # @UnusedVariable
+        # release allocated memory
+        gcf_io.free_GcfFile(obj)
+    return is_gcf
+
+
+def _read_gcf(filename, headonly=False, network='', station='',
+              locationcode='', bandcode="H", instrumentcode="H",
+              channel_prefix=None, blockmerge=True, cleanoverlap=True,
+              errorret=False, **kwargs):
     """
-    Reads a GCF file and returns a Stream object.
+    Reads a GCF file and returns a :class:`~obspy.core.stream.Stream`
 
-    only GCF files containing data records are supported.
+    Only GCF data records are supported. Function supports format as described
+    by GCF Reference
+    `SWA-RFC-GCFR Issue F, December 2021
+    <https://www.guralp.com/apps/ok?doc=GCF_format>`
 
     .. warning::
-        This function should NOT be called directly, it registers via the
-        ObsPy :func:`~obspy.core.stream.read` function, call this instead.
+            This function should NOT be called directly, it registers via the
+            ObsPy :func:`~obspy.core.stream.read` function, call this instead.
 
     :type filename: str
-    :param filename: GCF file to be read.
+    :param filename: path to GCF file to read
     :type headonly: bool, optional
-    :param headonly: If True read only head of GCF file.
+    :param headonly: if True only read block headers
+    :type network: str, optional
+    :param network: network code to use
+    :type station: str, optional
+    :param station: station code to use, if not specified unitID in gcf header
+       will be used (first 4 characters in stream_id)
+    :type locationcode: str, optional
+    :param locationcode: location code to use
```

Comment:
I'll rename this `location` to be more in line with rest of our code base

Reply:
Also, I kinda want to get rid of having a default channel_prefix for users. They should be responsible for using a sensible channel prefix themselves, otherwise people can end up with a misleading one set by our default

Reply:
Totally agree, however the existing gcf reader in obspy do use `HH` as default for `channel_prefix` ergo the failing test that are all due removing this. I kept it in order to not mess up code using obspy and relying on this.

Reply:
fair enough, I did not check that, I guess we can leave the default then :+1: